app/controllers/image_controller.rb

```diff
@@ -10,6 +10,19 @@ def index
     end
   end
 
+  def show
+    image = ProfileImage.find_by(id: params[:id])
+    if image
+      if image.cattle.user_id == current_user.id
+        render json: { image: image.fetch_image }, status: :ok
+      else
+        render json: { errors: ['Cattle does not belongs to you'] }, status: :unauthorized
```

Comment:
It's always nice to have explicit error messages specially that the error reducer can display errors without specifying its origin.